genesyscloud/provider/provider.go

```diff
@@ -244,10 +245,66 @@ func InitClientConfig(data *schema.ResourceData, version string, config *platfor
 	oauthclientID := data.Get("oauthclient_id").(string)
 	oauthclientSecret := data.Get("oauthclient_secret").(string)
 	basePath := GetRegionBasePath(data.Get("aws_region").(string))
+	config.BasePath = basePath
 
-	sdkDebugFilePath := data.Get("sdk_debug_file_path").(string)
+	diagErr := setUpSDKLogging(data, config)
+	if diagErr != nil {
+		return diagErr
+	}
+	setupProxy(data, config)
 
-	config.BasePath = basePath
+	config.AddDefaultHeader("User-Agent", "GC Terraform Provider/"+version)
+	config.RetryConfiguration = &platformclientv2.RetryConfiguration{
+		RetryWaitMin: time.Second * 1,
+		RetryWaitMax: time.Second * 30,
+		RetryMax:     20,
+		RequestLogHook: func(request *http.Request, count int) {
+			if count > 0 && request != nil {
+				log.Printf("Retry #%d for %s %s", count, request.Method, request.URL)
+			}
+		},
+		ResponseLogHook: func(response *http.Response) {
+			if response.StatusCode < http.StatusOK || response.StatusCode >= http.StatusMultipleChoices {
+				log.Printf("Response %s for request:%s %s", response.Status, response.Request.Method, response.Request.URL)
+			}
+		},
+	}
+
+	if accessToken != "" {
+		log.Print("Setting access token set on configuration instance.")
+		config.AccessToken = accessToken
+	} else {
+		config.AutomaticTokenRefresh = true // Enable automatic token refreshing
+
+		return withRetries(context.Background(), time.Minute, func() *retry.RetryError {
+			err := config.AuthorizeClientCredentials(oauthclientID, oauthclientSecret)
+			if err != nil {
+				if !strings.Contains(err.Error(), "Auth Error: 400 - invalid_request (rate limit exceeded;") {
+					return retry.NonRetryableError(fmt.Errorf("failed to authorize Genesys Cloud client credentials: %v", err))
+				}
+				return retry.RetryableError(fmt.Errorf("exhausted retries on Genesys Cloud client credentials. %v", err))
+			}
+
+			return nil
+		})
+	}
+
+	log.Printf("Initialized Go SDK Client. Debug=%t", data.Get("sdk_debug").(bool))
+	return nil
+}
+
+func withRetries(ctx context.Context, timeout time.Duration, method func() *retry.RetryError) diag.Diagnostics {
```

Comment:
I had to replicate the `withRetries` function here because importing it from utils caused an import cycle